components/ui/Alert.tsx

```diff
@@ -19,47 +19,143 @@
  */
 import * as classNames from 'classnames';
 import * as React from 'react';
+import styled from 'styled-components';
 import { translate } from '../../helpers/l10n';
 import Tooltip from '../controls/Tooltip';
 import AlertErrorIcon from '../icons/AlertErrorIcon';
 import AlertSuccessIcon from '../icons/AlertSuccessIcon';
 import AlertWarnIcon from '../icons/AlertWarnIcon';
 import InfoIcon from '../icons/InfoIcon';
-import ThemeContext from '../ThemeContext';
-import './Alert.css';
+import { SonarUiCommonTheme, SonarUiCommonThemeConsumer } from '../theme';
 
-type AlertDisplay = 'banner' | 'block' | 'inline';
+type AlertDisplay = 'banner' | 'inline' | 'block';
 type AlertVariant = 'error' | 'warning' | 'success' | 'info';
 
-export interface AlertProps {
+export interface Props {
   display?: AlertDisplay;
   variant: AlertVariant;
 }
 
-export function Alert(props: AlertProps & React.HTMLAttributes<HTMLDivElement>) {
+export function Alert(props: Props & React.HTMLAttributes<HTMLDivElement>) {
+  const isInline = () => props.display === 'inline';
+  const isBanner = () => props.display === 'banner';
+  const getAlertVariantInfo = (theme: SonarUiCommonTheme) => {
```

Comment:
This could be extracted from the `function Alert` component, adding the variant as a parameter.

Reply:
See fixup commit